ludwig/data/dataset/ray.py

```diff
@@ -100,8 +110,17 @@ def pipeline(
             shuffle: If true, the entire dataset is shuffled in memory before batching.
             fully_executed: If true, force full evaluation of the Ray Dataset by loading all blocks into memory.
             window_size_bytes: If not None, windowing is enabled and this parameter specifies the window size in bytes
-                    for the dataset.
+                    for the dataset. If None and the dataset is large, set to the window size determined at init.
         """
+        # If the user does not supply a window size and the dataset is large,
+        # set the window size to `<available memory> // 5`.
+        if self.auto_window and window_size_bytes is None:
+            ds_memory_size = self.in_memory_size_bytes
+            cluster_memory_size = ray.cluster_resources()["object_store_memory"]
+            if ds_memory_size > cluster_memory_size // 5:
+                logger.info("Dataset size is larger than object store, enabling windowed shuffle.")
```

Comment:
@jeffkinnison I don't think this is actually an accurate log message, since it's technically not larger, but it's greater than 20% of the object store memory. I'm wondering if we can change this to something like:

```
In-memory dataset size is greater than 20% of object store memory. Enabling windowed shuffling of data to prevent chances of OOMs. 
Read more here: <link to what windowing is, how to prevent this from happening, and the modeling performance impact>
```

And then we can add a section on windowed shuffling and its impact to the Ludwig docs. What do you think?

Reply:
That sounds good, do you have any suggestions for where to add the auto windowing info to the docs?

Reply:
Added the updated logging message in [24f36f5](https://github.com/ludwig-ai/ludwig/pull/2721/commits/24f36f5829d2197a39ff847dc016fd2108341052)